### ISSUE TITLE
Bug: TS bug when passing `data` in `useAssets`

### ISSUE THREAD

**Issue by itsezc** [pixijs/pixi-react#526]
### Current Behavior

When `data` is passed for `ResourceOptions`, TS errors are generated: 

```
ts: Type '{ resolution: number; resourceOptions: { scale: number; }; } | null' is not assignable to type 'Texture | undefined'.
  Type 'null' is not assignable to type 'Texture | undefined'.
```

### Expected Behavior

Types to be valid as a `Texture` 

### Steps to Reproduce

Pass `data` as:

```ts
const {
  assets: [texture],
  isSuccess
} = useAssets([{
  src: `/${name}.png`,
  data: ResourceDataOptions
}]);
```

### Environment

- **`@pixi/react` version**: v8 beta 9
- **`pixi.js` version**: v8
- **`React` version**: 19
- **`ReactDOM` version**: 19
- **Browser & Version**: Chrome 127
- **OS & Version**: MacOS 14

**Comment by thejustinwalsh** [comment]
If you do not explicitly hint useAssets with a type, it has no idea what type it should be.

Given the following deconstruction of your code back into the Assets.load call directly...
<img width="689" alt="image" src="https://github.com/user-attachments/assets/ec80088b-f689-4492-b846-54a1f5e69129">

When I attempt to translate that to a hook, I get the following, which I agree is incorrect; however, I'm afraid I have to disagree that it should be a Texture, as we did not tell the hook it was a Texture using the generic type param.
<img width="784" alt="image" src="https://github.com/user-attachments/assets/057e5e66-641c-433d-932b-2bf7435dac2c">

If I take a pass at typing this against Assets.load, I get what I believe are the results you are expecting.
<img width="1327" alt="image" src="https://github.com/user-attachments/assets/cbfae4ae-22ab-49e6-819e-b8b438ff05f4">

Finally, if I pass the type information to useAssets, I get the following, which, in my opinion, are the expected results with the exception of the data type being inferred incorrectly.
<img width="1223" alt="image" src="https://github.com/user-attachments/assets/16da7a7f-b31b-4efe-9921-766a658dfc81">


I believe the bug and the solution here is to ensure data is always `any` and not inferred from T as it is in the type implementation for pixi now. LMK if that is not you expect here.

**Comment by itsezc** [comment]
It's less of the what type data should be, while `data` should still be typed (where possible). It's more the return type of the return type of `assets: [texture]`, which isn't typed as `Texture` when passing `data`.

**Comment by thejustinwalsh** [comment]
It will not be typed as Texture unless you type it as so with `useAssets<Texture>`. You can not infer that it is a texture from a src url.

**Comment by thejustinwalsh** [comment]
I am working through some tests now and almost have a PR ready that will allow for proper typing.

**Comment by trezy** [comment]
@itsezc can you verify this issue is resolved in `@pixi/react@v8.0.0-beta.10`?

**Comment by itsezc** [comment]
> @itsezc can you verify this issue is resolved in `@pixi/react@v8.0.0-beta.10`?

Yes, its resolved
